docs/usage.md

```diff
@@ -42,7 +42,7 @@ To determine the project's dependencies, _deptry_ will scan the directory it is
     - development dependencies from `[tool.uv.dev-dependencies]` section and from the groups under `[project.optional-dependencies]` passed via the [`--pep621-dev-dependency-groups`](#pep-621-dev-dependency-groups) argument.
 1. If a `pyproject.toml` file with a `[project]` section is found, _deptry_ will assume it uses [PEP 621](https://peps.python.org/pep-0621/) for dependency specification and extract:
     - dependencies from `[project.dependencies]` and `[project.optional-dependencies]`.
-    - development dependencies from the groups under `[project.optional-dependencies]` passed via the [`--pep621-dev-dependency-groups`](#pep-621-dev-dependency-groups) argument.
```

Comment:
Note that although we do handle it in the implementation, I did not mention `[dependency-groups]` parsing for uv and PDM because they do not explicitly support PEP 735 yet. So although we assume that they support it, I'd prefer to not mention it for now until they explicitly add support for it, to avoid confusing users. Once support is added explicitly by one of the tool, we could update the documentation.